docs/src/main/asciidoc/lra-guide.adoc

```diff
@@ -0,0 +1,196 @@
+participant enlistment and it therefore must be available (for example if it or the
+network fail then something in the environment is responsible for restarting
+the coordinator or for repairing the network, respectively). To fulfil this task the
+coordinator must have access to durable storage for its logs (via a filesystem or in
+a database).  At the time of writing, managing coordinators is the responsibility of
+the user. An "out-of-the-box" solution will be forthcoming.
+
+== Examples
+
+The following is a simple example
+(https://github.com/eclipse/microprofile-lra/blob/master/spec/src/main/asciidoc/microprofile-lra-spec.adoc#java-annotations[taken from the specification document])
+of how to start an LRA and how to receive a notification when the LRA is later
+cancelled (the `@Compensate` annotated method is called) or closed (`@Complete` is called):
+
+[source,java]
+----
+@Path("/")
+@ApplicationScoped
+public class SimpleLRAParticipant
+{
+    @LRA(LRA.Type.REQUIRES_NEW)
+    @Path("/work")
+    @PUT
+    public void doInTransaction(@HeaderParam(LRA_HTTP_CONTEXT_HEADER) URI lraId)
+    {
+        /*
+         * Perform business actions in the context of the LRA identified by the
+         * value in the injected JAX-RS header. This LRA was started just before
+         * the method was entered (REQUIRES_NEW) and will be closed when the
+         * method finishes at which point the completeWork method below will be
+         * invoked.
+         */
+    }
+
+    @Complete
+    @Path("/complete")
+    @PUT
+    public Response completeWork(@HeaderParam(LRA_HTTP_CONTEXT_HEADER) URI lraId,
+                                 String userData)
+    {
+        /*
+         * Free up resources allocated in the context of the LRA identified by the
+         * value in the injected JAX-RS header.
+         *
+         * Since there is no @Status method in this class, completeWork MUST be
+         * idempotent and MUST return the status.
+         */
+         return Response.ok(ParticipantStatus.Completed.name()).build();
+    }
+
+    @Compensate
+    @Path("/compensate")
+    @PUT
+    public Response compensateWork(@HeaderParam(LRA_HTTP_CONTEXT_HEADER) URI lraId,
+                                   String userData)
+    {
+        /*
+         * The LRA identified by the value in the injected JAX-RS header was
+         * cancelled so the business logic should compensate for any actions
+         * that have been performed while running in its context.
+         *
+         * Since there is no @Status method in this class, compensateWork MUST be
+         * idempotent and MUST return the status
+         */
+         return Response.ok(ParticipantStatus.Compensated.name()).build();
+    }
+}
+----
+
+The example also shows that when an LRA is present its identifier can be obtained
+by reading the request headers.
+
+The same section of the specification document also includes an
```

Comment:
I found it rather odd to refer to the spec. I would not expect our users to read it (they won't :-)). 
So, I would rephrase to provide examples.

Reply:
okay, although isn't it still worth mentioning that the spec and TCK contain many other examples?

Reply:
Yes, definitely mention it, but your narrative should not expect readers to look at it.

Reply:
Nowhere does the narrative require readers to read the spec. The narrative only said that the example was taken from the spec.